modules/openapi-generator/src/main/resources/csharp-netcore/Configuration.mustache

```diff
@@ -94,7 +94,7 @@ namespace {{packageName}}.Client
         {
             UserAgent = "{{#httpUserAgent}}{{.}}{{/httpUserAgent}}{{^httpUserAgent}}OpenAPI-Generator/{{packageVersion}}/csharp{{/httpUserAgent}}";
             BasePath = "{{{basePath}}}";
-            DefaultHeader = new {{^net35}}Concurrent{{/net35}}Dictionary<string, string>();
+            DefaultHeaders = new {{^net35}}Concurrent{{/net35}}Dictionary<string, string>();
```

Comment:
@mario-d-s I suggest we keep `DefaultHeader` as switching to `DefaultHeaders` will make this PR a breaking changes (without fallback), which means we will need to target this change to 5.0.x branch instead (scheduled to be released next year May)

Reply:
As a matter of fact I _do_ keep the `DefaultHeader` around and loop it through the new `DefaultHeaders` (both `get` and `set`), but marked it as obsolete ([see](#diff-a166220228f92ff881905c32eade1d60R159)). I subsequently adjusted the rest of the files from using the obsolete property to the new one.

That is not to say I might not have missed a place where this **is** a breaking change, in that case please let me know but it should be safe AFAIK.

Reply:
Let's go with what you've and see if the community (users) spot anything we missed.